### PR TITLE
WIP - improve auto-resync

### PR DESCRIPTION
**Pull request by ObiWahn**
Please note that for legal reasons we require you to sign the 
[Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

File: arangod/Replication/TailingSyncer.cpp [hunk 1]
```diff
@@ -1177,7 +1177,10 @@ Result TailingSyncer::runInternal() {
   if (res.fail()) {
     // stop ourselves
     if (res.is(TRI_ERROR_REPLICATION_START_TICK_NOT_PRESENT) ||
+        res.is(TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND) || // data source -- collection or view
```
Comment (ObiWahn) [inline arangod/Replication/TailingSyncer.cpp]: This is the real change.

**Comment by ObiWahn** [comment]
Most of the changes are a result of the following comment `// TODO: merge with removeState` (https://github.com/arangodb/arangodb/pull/8020/files#diff-287322cb38782d8b016907e4a141e75eL1196)

File: arangod/Replication/TailingSyncer.cpp [hunk 2]
```diff
@@ -1189,33 +1192,15 @@ Result TailingSyncer::runInternal() {
...
-            << ", lastAppliedContinuousTick: " << _applier->_state._lastAppliedContinuousTick
-            << ", safeResumeTick: " << _applier->_state._safeResumeTick;
-
-        _applier->_state._lastProcessedContinuousTick = 0;
-        _applier->_state._lastAppliedContinuousTick = 0;
-        _applier->_state._safeResumeTick = 0;
-        _applier->_state._failedConnects = 0;
-        _applier->_state._totalRequests = 0;
-        _applier->_state._totalFailedConnects = 0;
-        _applier->_state._totalResyncs = 0;
```
Comment (danielhlarkin) [inline arangod/Replication/TailingSyncer.cpp]: Not all of these variables are zeroed out by the `ReplicationApplierState::reset` call inside `resetState`, and that call zeroes out some other variables as well. I'm less than 100% sure that we want the different set of variables cleared. Reasoning?

**Comment by ObiWahn** [comment]
> Hold on is this stopping the resync?

Yes if `res.fail()` is true the sync seems to stop. With the exclusion of the previously 2 now 3 results that trigger a reset.

**Comment by ObiWahn** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2847/

**Comment by ObiWahn** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2950/

**Comment by ObiWahn** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3131/

**Pull request merged**